azavea/climate-change-lab
# Feature/kak/staging site#150

## Overview

Set up lab staging site and publish there via Jenkins on push to `develop`.


### Notes

Copied configurations for S3 bucket, CloudFront configuration, and Route 53 DNS generally from the production site configurations. @sharph might want to review the AWS service configurations.

By default, `s3_website push` will now publish to staging. Production site s3 configuration now lives in a non-default location, so it may still be easily published from within the VM by specifying the configuration directory as an extra argument.

Jenkins configuration has been updated to use staging for the API host. After merge, we should check that Jenkins has published properly.

## Testing Instructions

 * `vim src/app/constants.ts`
 * Change `apiHost` to https://app.staging.climate.azavea.com
 * `npm run build:prod`
 * Test publishing to staging: `s3_website push`
 * Should say: `Successfully pushed the website to http://lab.staging.climate.azavea.com.s3-website-us-east-1.amazonaws.com`
 * `vim src/app/constants.ts`
 * Change `apiHost` to https://app.climate.azavea.com
 * `npm run build:prod`
 * Test publishing to production: `s3_website push --config-dir=production_s3/`
 * Should say: `Successfully pushed the website to http://lab.climate.azavea.com.s3-website-us-east-1.amazonaws.com`
 * Both production and staging sites should be up and querying the correct host: https://lab.staging.climate.azavea.com and https://lab.climate.azavea.com

Closes #150
